src/nprintml/pipeline.py

```diff
@@ -0,0 +1,341 @@
+"""nprintML pipeline framework
```

Comment:
Obviously this framework might have to change as we figure the library out; and, it might be overkill for what we're doing. But, I wanted to make it so that the CLI needn't know anything about the pipeline -- that the steps just configure themselves. I think, so far, this will do a decent job, at that.